modules/dop/providers/guide/provider.go

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) 2021 Terminus, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package guide
+
+import (
+	"context"
+
+	"github.com/jinzhu/gorm"
+
+	"github.com/erda-project/erda-infra/base/logs"
+	"github.com/erda-project/erda-infra/base/servicehub"
+	"github.com/erda-project/erda-infra/pkg/transport"
+	"github.com/erda-project/erda-infra/providers/i18n"
+	"github.com/erda-project/erda-proto-go/dop/guide/pb"
+	"github.com/erda-project/erda/bundle"
+	"github.com/erda-project/erda/modules/dop/conf"
+	"github.com/erda-project/erda/modules/dop/dao"
+	"github.com/erda-project/erda/modules/dop/providers/guide/db"
+	"github.com/erda-project/erda/pkg/common/apis"
+	"github.com/erda-project/erda/pkg/cron"
+	"github.com/erda-project/erda/pkg/database/dbengine"
+)
+
+type config struct {
+}
+
+type provider struct {
+	Cfg      *config
+	Log      logs.Logger
+	bundle   *bundle.Bundle
+	DB       *gorm.DB           `autowired:"mysql-client"`
+	Register transport.Register `autowired:"service-register" required:"true"`
+	Trans    i18n.Translator    `translator:"project-pipeline" required:"true"`
+
+	GuideService *GuideService
+}
+
+func (p *provider) Run(ctx context.Context) error {
+	go func() {
```

Comment:
`provider.Run` also support blocking func, thus async is not necessary, @sfwn can you help to confirm it?